src/types/bytes.rs

```diff
@@ -26,6 +26,21 @@ impl PyBytes {
         unsafe { py.from_owned_ptr(ffi::PyBytes_FromStringAndSize(ptr, len)) }
     }
 
+    /// Creates a new Python bytestring object.
+    /// The uninitialised bytestring must be initialised in the `init` closure.
+    ///
+    /// Panics if out of memory.
+    pub fn new_with<F: Fn(&mut [u8])>(py: Python<'_>, len: usize, init: F) -> &PyBytes {
+        let length = len as ffi::Py_ssize_t;
+        let pyptr = unsafe { ffi::PyBytes_FromStringAndSize(std::ptr::null(), length) };
```

Comment:
As this and the next 4 lines all contain unsafe, could perhaps just refactor to be a single unsafe block?

Reply:
Fixed in 20d27ff050b57a5ceb446c881efaf301b97b4cda